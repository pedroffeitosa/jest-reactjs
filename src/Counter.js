import React from 'react';

function Counter({ count }) {
  return (
    <div className="Counter">
      <h1>VocÃª tem {count} counters.</h1>
    </div>
  );
}

export default Counter;
