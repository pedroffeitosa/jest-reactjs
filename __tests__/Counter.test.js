import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import { shallow } from 'enzyme';
import Counter from '../src/Counter';

describe('Testing Counter Component',()=>{
  it('should render correctly', ()=> {
    const wrapper = shallow(<Counter count={5}/>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
    wrapper.setProps({ count: 10 });
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  })
})