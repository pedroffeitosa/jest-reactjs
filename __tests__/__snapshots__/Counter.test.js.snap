// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Counter Component should render correctly 1`] = `
<div
  className="Counter"
>
  <h1>
    Você tem 
    5
     counters.
  </h1>
</div>
`;

exports[`Testing Counter Component should render correctly 2`] = `
<div
  className="Counter"
>
  <h1>
    Você tem 
    10
     counters.
  </h1>
</div>
`;
